62_CAD/DrawResourcesFiller.h

```diff
@@ -24,6 +25,24 @@ struct DrawBuffers
 	smart_refctd_ptr<BufferType> lineStylesBuffer;
 };
 
+// TODO: Better place for this
+enum class MsdfFillPattern: uint32_t
```

Comment:
This is more high-level than DrawResourcesFiller, so make that and the factory functions that give you msdfs into a seperate file. this way we can easily copy paste into n4ce and use that there.

Reply:
We'll have classes/structs related to text placement and msdf generations anyways, it makes sense to put them all in there along with the hashing of the data

Reply:
When doing it one by one, the validation layer kept claiming that the resource was on the wrong layout, we could make it transition from shader read, and only use undefined for the first transition

Reply:
UNDEFINED is not a wrong layout.
Your next barrier after the upload was the issue. Because some of the textures were freshly TRANSFER_DST and some of them were old SHADER_READONLY
Doing both the before and after barriers for only the uploading ones shouldn't result in a validation error. 
Any transition barrier chain which starts from UNDEFINED, should be pretty straightforward because you're completely overwriting the previous slot

Reply:
Done (I think commented on the wrong place here)